charms/benchmark/__init__.py

```diff
@@ -118,9 +120,26 @@ def start():
         If the cabs-collector charm is installed, take a snapshot
         of the current profile data.
         """
+        # Do profile data collection immediately on this unit
         if os.path.exists(COLLECT_PROFILE_DATA):
             subprocess.check_output([COLLECT_PROFILE_DATA])
 
+        # Tell the benchmark-gui charm the action_id via the benchmark
+        # relation. Benchmark-gui will pass the action_id to all
+        # collector charms in the environment (via the collector relation),
+        # triggering profile data collection on each.
+        if os.environ['CHARM_DIR']:
```

Comment:
this will fail if CHARM_DIR isn't present, it's best to do `if 'CHARM_DIR in os.environ` or to use `os.environ.get('CHARM_DIR', False)`